src/parser.rs

```diff
@@ -61,8 +61,7 @@ impl<'a> Parser<'a> {
                         let end = self.advance_while(is_symbol_tail);
                         Ok(Value::Symbol(self.str[start..end].into()))
                     }
-                    None | Some(' ') | Some('\t') | Some('\n') => Ok(Value::Symbol(ch.to_string())),
-                    _ => unimplemented!(),
+                    None | Some(_) => Ok(Value::Symbol(ch.to_string()))
```

Comment:
I don't think this change is right. With this, wouldn't `+µ` for example parse as `Symbol("+")`?

Reply:
It depends what is considered "right".

The EDN spec limits symbols to a very small list of characters:

> Symbols begin with a non-numeric character and can contain alphanumeric characters and . * + ! - _ ? $ % & = < >. If -, + or . are the first character, the second character (if any) must be non-numeric. Additionally, : # are allowed as constituent characters in symbols other than as the first character.

By this logic, `+` is a valid symbol character, while `µ` is not, and therefore it should parse to `Symbol("+")`, followed by `Err("invalid leading character: µ")` on the next `read()`.

However, the official EDN parser will happily make a symbol out of nearly any Unicode character:

```
=> (type (clojure.edn/read-string "µ"))
clojure.lang.Symbol
=> (type (clojure.edn/read-string "🔥"))
clojure.lang.Symbol
=> (type (clojure.edn/read-string "\0"))
clojure.lang.Symbol
```

So the question is, which is right? The spec, or the reference implementation?

Reply:
I brought this question up in the Clojurians Slack, and people seemed to recommend following the spec over the behavior of the Clojure parser implementation: https://clojurians.slack.com/archives/C03S1KBA2/p1533848073000136